src/download.ts

```diff
@@ -73,14 +76,25 @@ export async function downloadPage(title: string, link: string): Promise<void> {
 
     await page.addStyleTag({ content: 'div[class^="styles__PrevNextButtonWidgetStyled"], div[class^="styles__Footer"], nav { display: none !important; }' });
 
-    await page.evaluate(({ SAVE_AS, SAVE_LESSON_AS }) => {
+    const languages = await page.evaluate(({ SAVE_AS, SAVE_LESSON_AS }) => {
       // Expand all slides in the page
       const xPathResult = document.evaluate('//button[contains(@class, "AnimationPlus")]', document, null, XPathResult.ORDERED_NODE_SNAPSHOT_TYPE, null);
       for (let i = 0; i < xPathResult.snapshotLength; i++) {
         const element = xPathResult.snapshotItem(i) as HTMLElement;
         element.click();
       }
 
+      const languages = [];
+      const codeContainer = document.getElementsByClassName('code-container');
+      if (codeContainer.length === 0) {
+        return languages;
+      }
+
+      const languageTabs = (codeContainer[0].previousSibling.firstChild as HTMLElement).querySelectorAll('span.desktop-only');
```

Comment:
We need to handle failure case for this
`const previousSibling  = codeContainer[0].previousSibling;
      if  (previousSibling == null) {
        return languages;
      }

      const languageTabs = (previousSibling.firstChild as HTMLElement).querySelectorAll('span.desktop-only');
      languageTabs.forEach((e) => {
        languages.push(e.querySelector('span').innerText);
      });`